src/tunnel.cc

```diff
@@ -191,6 +176,9 @@ class TunnelStateData: public PeerSelectionInitiator
     // TODO: remove after fixing deferred reads in TunnelStateData::copyRead()
     CodeContext::Pointer codeContext; ///< our creator context
 
+    /// a cached ClientHttpRequest::clientExpectsConnectResponse() value at the object creation time
+    bool clientExpectsConnectResponse;
```

Comment:
To highlight that this information is about the state in the past (and to avoid uninitialized values):

```suggestion
    bool clientExpectedConnectResponse = false;
```

Alternatively, we can keep the name and reset the data member to false after we decide that the client no longer expects that response, but I think keeping this data member (essentially) constant (as in the current PR code) is better overall because the (explicit) state is simpler that way.